src/js/utils/throttle.js

```diff
@@ -0,0 +1,32 @@
+
+/**
+ * Throttle for functions that return a promise. If additional calls are made
+ * while the throttle is on, the original promise will be returned.
+ *
+ * Usage:
+ *
+ * const func = () => Promise.resolve('foo')
+ *
+ * const throttle = new Throttle();
+ * const throttledFunc = () => throttle.exec(() => func());
+ * throttledFunc().then((foo) => {})
+ */
+export class Throttle {
```

Comment:
3rd party promise throttles do exist, but I feel it's below the complexity threshold to justify adding a dependency.